# Bài 1 - Tính giá cổ phiếu

 -- Nhập dữ liệu giá cổ phiếu trong 8 ngày

In [9]:
gia_theo_ngay = [10.1, 9.5, 10.7, 10.8, 10.6, 10.0, 10.2, 9.5]

-- Tạo hàm tính toán mức độ tăng giá hoặc giảm giá

In [28]:
def TinhGiaTangGiam(gia_ngay,index):
    if(index == 0):
        print(f'Giá Ngày Đầu: {gia_theo_ngay[index]}')
        print('-----------')
    elif (index != 0):
        if((gia_ngay - gia_theo_ngay[index-1]) / gia_theo_ngay[index-1]) > 0.07:
            print(f'Bán Giá Hôm Nay: {gia_ngay}')
            print("Mức Tăng : ", abs((gia_ngay - gia_theo_ngay[index-1]) / gia_theo_ngay[index-1])*100,"%")
            print("----------")
        elif ((gia_theo_ngay[index -1] - gia_ngay) / gia_theo_ngay[index-1]) > 0.06:
            print(f'Mua Giá Hôm Nay: {gia_ngay}')
            print("Mức giảm: ",abs((gia_theo_ngay[index-1] - gia_ngay) / gia_theo_ngay[index-1])*100,"%")
            print('-----------')
        else: 
            print(f'Không Mua Bán Gì Hôm Nay {gia_ngay}')
            print("Mức chênh: ",abs((gia_ngay - gia_theo_ngay[index-1]) / gia_theo_ngay[index-1])*100,"%")
            print('-----------')

-- Chạy vòng lặp để kiểm tra giá từng ngày

In [29]:
for i,val in enumerate(gia_theo_ngay):
    TinhGiaTangGiam(val,i)
    

Giá Ngày Đầu: 10.1
-----------
Không Mua Bán Gì Hôm Nay 9.5
Mức chênh:  5.940594059405938 %
-----------
Bán Giá Hôm Nay: 10.7
Mức Tăng :  12.631578947368412 %
----------
Không Mua Bán Gì Hôm Nay 10.8
Mức chênh:  0.9345794392523498 %
-----------
Không Mua Bán Gì Hôm Nay 10.6
Mức chênh:  1.8518518518518614 %
-----------
Không Mua Bán Gì Hôm Nay 10.0
Mức chênh:  5.660377358490563 %
-----------
Không Mua Bán Gì Hôm Nay 10.2
Mức chênh:  1.9999999999999927 %
-----------
Mua Giá Hôm Nay: 9.5
Mức giảm:  6.862745098039209 %
-----------


# Bài 2 - Tính BHXH và PIT

-- Công thức tính thuế : https://luatvietnam.vn/thue-phi/file-excel-tinh-thue-thu-nhap-ca-nhan-565-23632-article.html

In [44]:
def TinhBHXH_PIT(wage,bonus):
    #Công thức tính BHXH
    bhxh = wage * 0.08
    #Công thức thu nhập tính thuế
    thu_nhap_tinh_thue = (wage + bonus) - bhxh
    pit = 0
    if(thu_nhap_tinh_thue < 5000000):
        pit = thu_nhap_tinh_thue * 0.05
    elif(5000000<= thu_nhap_tinh_thue < 10000000):
        pit = thu_nhap_tinh_thue * 0.1 - 250000
    elif (10000000<= thu_nhap_tinh_thue < 18000000):
        pit = thu_nhap_tinh_thue * 0.15 - 750000
    elif (18000000<= thu_nhap_tinh_thue < 32000000):
        pit = thu_nhap_tinh_thue * 0.2 - 1650000
    elif (32000000<= thu_nhap_tinh_thue < 52000000):
        pit = thu_nhap_tinh_thue * 0.25 - 3250000
    elif (52000000<= thu_nhap_tinh_thue < 80000000):
        pit = thu_nhap_tinh_thue * 0.3 - 5850000
    else :pit = thu_nhap_tinh_thue * 0.35 - 9850000    
    return {"Bảo hiểm xã hội":bhxh, "Thuế thu nhập cá nhân":pit}, 

In [43]:
TinhBHXH_PIT(10000000,5000000)

({'Bảo hiểm xã hội': 800000.0, 'Thuế thụ nhập cá nhân': 1380000.0},)

# Bài Tập Tính Tần Suất, Tần Số, Mean và Median

In [47]:
import csv
csv_file_1 = r"C:\Users\NGUYEN ANH TUAN\Desktop\Cybersoft-DataAnalyticWithPython\files\example_bai_1.2.csv"

In [48]:
# Tách các cột thành từng list riêng
list_rating = []
list_city = []
list_gender =  []
list_amount = []

with open(csv_file_1, 'r', encoding="utf-8") as ds:
    data_reader = csv.reader(ds)
    for row in data_reader:
        list_city.append(row[0])
        list_gender.append(row[1])
        list_amount.append(row[2])
        list_rating.append(row[3])
        
# Loại bỏ header       
del(list_rating[0])
del(list_city[0])
del(list_gender[0])
del(list_amount[0])

# Tần suất khác mua hàng các thành phố

In [49]:
HN = 0
HCM = 0
DN = 0
for index, val in enumerate(list_city):
    if(val == "TP HCM"):
        HCM += 1
    elif (val == "Hà Nội"):
        HN += 1
    else : DN += 1
print(HN,HCM,DN)
print({"HN":round(HN/(HN+HCM+DN)*100,2)},{"HCM":round(HCM/(HN+HCM+DN)*100,2)},{"DN":round(DN/(HN+HCM+DN)*100,2)})

274 490 236
{'HN': 27.4} {'HCM': 49.0} {'DN': 23.6}


# Tần Suất Khách Nam và Nữ

In [50]:
Male = 0
Female = 0
for index, val in enumerate(list_gender):
    if(val == 'Nữ'):
        Female += 1
    elif(val == 'Nam'):
        Male += 1
print(f'Female:{Female}, Male:{Male}')
print({"Female":round(Female/(Female+Male)*100,2),"Male":round(Male/(Female+Male)*100,2)})

Female:501, Male:499
{'Female': 50.1, 'Male': 49.9}


# Median

In [51]:
list_amount_sorted = []
for val in list_amount:
    list_amount_sorted.append(float(val))
list_amount_sorted.sort()
list_amount_sorted

[10.6785,
 12.6945,
 13.167,
 13.419,
 14.679,
 16.107,
 16.2015,
 16.275,
 17.094,
 18.6375,
 19.194,
 19.2465,
 20.1075,
 20.685,
 22.386,
 22.659,
 23.499,
 23.751,
 24.108,
 25.263,
 26.25,
 26.5545,
 26.7225,
 26.733,
 26.796,
 28.4235,
 30.219,
 30.408,
 30.996,
 31.227,
 31.752,
 31.9305,
 32.1405,
 32.151,
 32.277,
 32.529,
 33.3585,
 33.432,
 33.495,
 33.936,
 34.629,
 35.196,
 35.3115,
 36.5505,
 37.611,
 37.6845,
 38.85,
 40.341,
 40.53,
 40.9605,
 41.076,
 41.391,
 41.454,
 41.7375,
 42.3675,
 43.869,
 44.352,
 44.5935,
 44.9925,
 45.108,
 45.927,
 46.683,
 47.859,
 48.51,
 48.7305,
 49.308,
 49.4235,
 49.77,
 49.812,
 51.0405,
 51.1455,
 53.151,
 53.34,
 53.928,
 54.0435,
 54.9675,
 54.999,
 55.041,
 55.2825,
 55.881,
 56.1225,
 56.406,
 56.469,
 56.952,
 57.078,
 57.1725,
 57.6975,
 57.876,
 58.2225,
 59.325,
 60.816,
 62.0025,
 63.2625,
 63.315,
 63.9135,
 63.9975,
 65.604,
 65.7405,
 66.402,
 66.8745,
 68.2395,
 68.985,
 69.09,
 69.111,
 69.405,
 69.6675,
 69.72,
 69.99

In [52]:
int(len(list_amount_sorted))

1000

-- Vì đồ dài list là 1000 nên median sẽ lấy trung bình của giá trị thứ 499 và 500

In [53]:
median_amount = (list_amount_sorted[int(len(list_amount_sorted)/2)] + list_amount_sorted[int(len(list_amount_sorted)/2) - 1])/2
median_amount

253.848

-- Median của Rating


In [54]:
list_rating_sorted = []
for val in list_rating:
    list_rating_sorted.append(float(val))
list_rating_sorted.sort()
list_rating_sorted

[4.0,
 4.0,
 4.0,
 4.0,
 4.0,
 4.0,
 4.0,
 4.0,
 4.0,
 4.0,
 4.0,
 4.1,
 4.1,
 4.1,
 4.1,
 4.1,
 4.1,
 4.1,
 4.1,
 4.1,
 4.1,
 4.1,
 4.1,
 4.1,
 4.1,
 4.1,
 4.1,
 4.1,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.2,
 4.3,
 4.3,
 4.3,
 4.3,
 4.3,
 4.3,
 4.3,
 4.3,
 4.3,
 4.3,
 4.3,
 4.3,
 4.3,
 4.3,
 4.3,
 4.3,
 4.3,
 4.3,
 4.4,
 4.4,
 4.4,
 4.4,
 4.4,
 4.4,
 4.4,
 4.4,
 4.4,
 4.4,
 4.4,
 4.4,
 4.4,
 4.4,
 4.4,
 4.4,
 4.4,
 4.5,
 4.5,
 4.5,
 4.5,
 4.5,
 4.5,
 4.5,
 4.5,
 4.5,
 4.5,
 4.5,
 4.5,
 4.5,
 4.5,
 4.5,
 4.5,
 4.5,
 4.6,
 4.6,
 4.6,
 4.6,
 4.6,
 4.6,
 4.6,
 4.6,
 4.7,
 4.7,
 4.7,
 4.7,
 4.7,
 4.7,
 4.7,
 4.7,
 4.7,
 4.7,
 4.7,
 4.7,
 4.8,
 4.8,
 4.8,
 4.8,
 4.8,
 4.8,
 4.8,
 4.8,
 4.8,
 4.8,
 4.8,
 4.8,
 4.8,
 4.9,
 4.9,
 4.9,
 4.9,
 4.9,
 4.9,
 4.9,
 4.9,
 4.9,
 4.9,
 4.9,
 4.9,
 4.9,
 4.9,
 4.9,
 4.9,
 4.9,
 4.9,
 5.0,
 5.0,
 5.0,
 5.0,
 5.0,
 5.0,
 5.0,
 5.0,
 5.0,
 5.0,
 5.0,
 5.0,
 5.0,
 5.0

In [55]:
int(len(list_rating_sorted))

1000

In [56]:
median_rating = (list_rating_sorted[int(len(list_rating_sorted)/2)] + list_rating_sorted[int(len(list_rating_sorted)/2) - 1])/2
median_rating

7.0

# MEAN

-- Rating

In [57]:
sum_rating = 0
avg_rating = 0
for index, val in enumerate(list_rating):
    sum_rating += float(val)
    avg_rating = sum_rating / len(list_rating)
avg_rating

6.972700000000003

-- Tổng Đơn

In [58]:
sum_amount = 0
avg_amount = 0
for index, val in enumerate(list_amount):
    sum_amount += float(val)
    avg_amount = sum_amount / len(list_amount)
avg_amount

322.96674900000005